src/hash/rescue/mod.rs

```diff
@@ -15,6 +15,11 @@ pub use rpx::{Rpx256, RpxDigest};
 #[cfg(test)]
 mod tests;
 
+#[cfg(target_feature = "avx2")]
+use crate::arch::x86_64_avx2::{
+    apply_inv_sbox as optimized_inv_sbox, apply_sbox as optimized_sbox,
+};
```

Comment:
Given that we now have quite a few feature-dependent functions, would it make sense to group them together into sub-modules. For example, we could do something like this:

```Rust
#[cfg(target_feature = "avx2")]
mod optimizations {
    use crate::arch::x86_64_avx2::{
        apply_inv_sbox as optimized_inv_sbox, apply_sbox as optimized_sbox,
    };

    fn optimized_add_constants_and_apply_sbox(
        state: &mut [Felt; STATE_WIDTH],
        ark: &[Felt; STATE_WIDTH],
    ) -> bool {
        super::add_constants(state, ark);
        unsafe {
            optimized_sbox(std::mem::transmute(state));
        }
        true
    }

    fn optimized_add_constants_and_apply_inv_sbox(
        state: &mut [Felt; STATE_WIDTH],
        ark: &[Felt; STATE_WIDTH],
    ) -> bool {
        super::add_constants(state, ark);
        unsafe {
            optimized_inv_sbox(std::mem::transmute(state));
        }
        true
    }
}
```

We could then do the same for SVE-based optimizations and also for no-optimizations case. The, we'd just import these functions at the top level as:
```Rust
use optimizations::{optimized_add_constants_and_apply_sbox, optimized_add_constants_and_apply_inv_sbox};
```

Reply:
I like this idea. Since I moved arch specific code to `rescue` module, it seemed fitting to add optimizations directly to `arch`.